pkg/install/deploystorage.go

```diff
@@ -103,12 +104,28 @@ func (i *Installer) deployStorageTemplate(ctx context.Context, installConfig *in
 
 		applications := graphrbac.NewApplicationsClient(spp.TenantID, spGraphAuthorizer)
 
-		res, err := applications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
+		timeoutCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+		defer cancel()
+		// NOTE: Do not override err with the error returned by wait.PollImmediateUntil.
+		// Doing this will not propagate the latest error to the user in case when wait exceeds the timeout
+		wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+			var res azgraphrbac.ServicePrincipalObjectResult
+			res, err = applications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
+			if err != nil {
+				if strings.Contains(err.Error(), "Authorization_IdentityNotFound") {
+					i.log.Info(err)
+					return false, nil
+				}
+
+				return false, err
```

Comment:
What should become of this non-`Authorization_IdentityNotFound` error here? It currently just disappears into the void.

Reply:
we will log it in reply() and return a 500, just like any other case

Reply:
I see. A non-`Authorization_IdentityNotFound` error would get ultimately returned on line 127 not _through_ `PollImmediateUntil` as implied by the return statement on 120, but because this `ConditionFunc` encloses `err`. The same result could be achieved if line 120 was `return false, nil`. Having that with a comment would be easier to read, I think.